aspnetcore/fundamentals/startup.md

```diff
@@ -16,57 +16,60 @@ uid: fundamentals/startup
+Any services added by the ``WebHostBuilder`` ``ConfigureServices`` method may be requested by the ``Startup`` class constructor or its ``Configure`` method. Use `WebHostBuilder` to provide any services you need during `Startup` methods.
 
-[!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
+## The Configure method
 
-Adding services to the services container makes them available within your application via [dependency injection](dependency-injection.md).
+The `Configure` method is used to specify how the ASP.NET application will respond to HTTP requests. The request pipeline is configured by adding [middleware](middleware.md) components to an `IApplicationBuilder` instance that is provided by dependency injection.
 
-## Services Available in Startup
+In the following example from the default web site template, several extension methods are used to configure the pipeline with support for [BrowserLink](http://vswebessentials.com/features/browserlink), error pages, static files, ASP.NET MVC, and Identity.
+
+[!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=8,9,10,14,17,19,21&start=58&end=84)]
 
-ASP.NET Core dependency injection provides application services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or one of its `Configure` or `ConfigureServices` methods. 
+Each `Use` extension method adds a [middleware](xref:fundamentals/middleware) component to the request pipeline. For instance, the `UseMvc` extension method adds the [routing](routing.md) middleware to the request pipeline and configures [MVC](xref:mvc/overview) as the default handler.
 
-Looking at each method in the `Startup` class in the order in which they are called, the following services may be requested as parameters:
+For more information about how to use `IApplicationBuilder`, see [Middleware](xref:fundamentals/middleware).
 
-* In the constructor:  `IHostingEnvironment`, `ILoggerFactory`
-* In the `ConfigureServices` method:  `IServiceCollection`
-* In the `Configure` method:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`, `IApplicationLifetime`
+Additional services, like `IHostingEnvironment` and `ILoggerFactory` may also be specified in the method signature, in which case these services will be [injected](dependency-injection.md) if they are available. 
 
 ## Additional Resources
 
-* [Working with Multiple Environments](environments.md)
-* [Middleware](middleware.md)
-* [Logging](logging.md)
-* [Configuration](configuration.md)
+* [Working with Multiple Environments](xref:fundamentals/environments)
+* [Middleware](xref:fundamentals/middleware)
+* [Logging](xref:fundamentals/logging)
+* [Configuration](xref:fundamentals/configuration)
+* [List All Services Available to an ASP.NET Core App](https://github.com/ardalis/AspNetCoreStartupServices)
```

Comment:
@ardalis  can you remove this for now? If @danroth27  approves it, we can add back in edit from browser PR.